libsql/internal/http/http.go

```diff
@@ -1,117 +1,206 @@
-		return nil, err
+	paramExtractor := &ParametersExtractorListener{}
+	antlr.ParseTreeWalkerDefault.Walk(paramExtractor, tree)
+	if paramExtractor.err != nil {
+		return params{}, paramExtractor.err
 	}
-	if resp.StatusCode != http.StatusOK {
-		var err_response struct {
-			Message string `json:"error"`
+
+	stmtParams := NewParams(queryParams.Type())
+
+	switch queryParams.Type() {
+	case positionalParameters:
+		if positionalParametersOffset+paramExtractor.positionalParamsCount > len(queryParams.positional) {
+			return params{}, fmt.Errorf("missing positional parameters")
+		}
+		stmtParams.positional = queryParams.positional[positionalParametersOffset : positionalParametersOffset+paramExtractor.positionalParamsCount]
+	case namedParameters:
+		stmtParametersNeeded := make(map[string]bool)
+		for _, stmtParametersName := range paramExtractor.nameParams {
+			stmtParametersNeeded[stmtParametersName] = true
 		}
-		if err := json.Unmarshal(body, &err_response); err != nil {
-			return nil, fmt.Errorf("failed to execute SQL: %s", sql)
+		for queryParamsName, queryParamsValue := range queryParams.named {
+			if stmtParametersNeeded[queryParamsName] {
+				stmtParams.named[queryParamsName] = queryParamsValue
+				delete(stmtParametersNeeded, queryParamsName)
+			}
 		}
-		return nil, fmt.Errorf("failed to execute SQL: %s\n%s", sql, err_response.Message)
 	}
 
-	type errObject struct {
-		Message string `json:"message"`
-	}
+	return stmtParams, nil
+}
 
-	var results []struct {
-		Results *resultSet `json:"results"`
-		Error   *errObject `json:"error"`
+func removeParamPrefix(paramName string) (string, error) {
+	if paramName[0] == ':' || paramName[0] == '@' || paramName[0] == '$' {
+		return paramName[1:], nil
 	}
-	if err := json.Unmarshal(body, &results); err != nil {
-		return nil, err
+	return "", fmt.Errorf("all named parameters must start with ':', or '@' or '$'")
+}
+
+type ParametersExtractorListener struct {
+	parser.BaseSQLiteParserListener
+
+	err error
+
+	nameParams            []string
+	positionalParamsCount int
+}
+
+func (pel *ParametersExtractorListener) EnterExpr(ctx *parser.ExprContext) {
+	if pel.err != nil {
+		return
 	}
-	if len(results) != 1 {
-		return nil, fmt.Errorf("wrong number of results for SQL: %s\nexpected 1 got %d", sql, len(results))
+	bindParameter := ctx.BIND_PARAMETER()
+	if bindParameter == nil {
+		return
 	}
-	if results[0].Error != nil {
-		return nil, fmt.Errorf("failed to execute SQL: %s\n%s", sql, results[0].Error.Message)
+
+	parameter := bindParameter.GetText()
+
+	if strings.HasPrefix(parameter, "?") {
+		pel.positionalParamsCount++
+	} else {
+		paramWithoutPrefix, err := removeParamPrefix(parameter)
+		if err != nil {
+			pel.err = err
+		} else {
+			pel.nameParams = append(pel.nameParams, paramWithoutPrefix)
+		}
 	}
-	if results[0].Results == nil {
-		return nil, fmt.Errorf("no results for SQL: %s", sql)
+}
+
+func splitStatementToPieces(statementsString string) (pieces []string) {
```

Comment:
I'm thinking about this. We only need to split the statements because HTTP request expect them to be sent a `statements` array and one statement per field. Thinking this way, maybe it should belong to http.go file. Right?